### DISCUSSION TITLE
Is it possible to connect to OCI Autonomous Database?

### DISCUSSION THREAD

**Discussion by shirakia** [oracle/python-oracledb#18] [Q&A] [answered]
Hi, this library is super cool since it doesn't require instant client and is compatible with M1 Mac.
But how to connect to Autonomous DBs?

I couldn't find nice docs.
More specifically, what should I put for `dsn` for `oracledb.connect` ?

**Comment by cjbj** [comment, marked as answer]
Here's the documentation: [Connecting to Oracle Cloud Autonomous Databases](https://python-oracledb.readthedocs.io/en/latest/user_guide/connection_handling.html#connecting-to-oracle-cloud-autonomous-databases) it has examples for both 1-way (walletless) and mTLS (with wallets).

**Reply by shirakia** [reply]
Thanks!!
It worked very nicely by that doc and this library.
